view/Requirements_Backend.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+interface Requirements_Backend {
```

Comment:
This interface is pretty heavily tied to our specific implementation of Requirements. I would expect that if we were to bring in another requirements processor (e.g. something that pushed more of the heavily lifting to a node-based asset pipeline) then it would be hard to fit into this interface, and would instead have its own methods.

Perhaps a tag like this

    <% require javascript("bla") %>

Should instead call:

    Injector::inst()->get('Requirements')->addRequirement('javascript', 'bla');

That way, different requiremnts backends can provided alternative semantics for the <% require %> tag.